### DISCUSSION TITLE
Allow custom headerbar colors when `window-theme=ghostty`

### DISCUSSION THREAD

**Discussion by jcollie** [ghostty-org/ghostty#2447] [Feature Requests, Ideas] [answered]
When you set `window-theme=ghostty` in your config, the headerbar colors are set to the terminal's `foreground` and `background`. This works well for _some_ colors, but not others (`background=#ffffff` in particular). I propose adding `headerbar-foreground` and `headerbar-background` (`null` by default). If set, and `window-theme=ghostty` these colors would be used instead of the terminal `foreground` and `background` to set the headerbar colors.

**Comment by mitchellh** [comment]
Is there a reason these have to be tied to `window-theme=ghostty`? It seems like providing perhaps a `gtk-headerbar-foreground` (and background) could just apply when set, and `window-theme=ghostty` is a shortcut for those defaulting to the foreground/background.

**Reply by jcollie** [reply]
I think tying it to some specific setting of `window-theme` keeps the logic of when the config applies a little more straightforward. Doesn't necessarily need to be `ghostty`, could be `custom` or something like that. I don't have a strong feelings if people would rather not. I also didn't add a `gtk-` prefix to the settings in case the macOS side wanted to do something similar.

**Reply by mitchellh** [reply]
I think starting with a `gtk-` prefix is going to be safer, I'm not sure if there is a timeline or intention to really support something like this on macOS and when we do we can deprecate the gtk prefixes if necessary.

I'll think about the `window-theme` thing...

**Comment by jcollie** [comment]
Had a thought today about a different way to configure headerbars/CSS on GTK. (This would require GTK 4.16 and libadwaita.) There would be a `window-theme=gtk` setting, and then one or more entries like this:

```
gtk-theme=headerbar-fg-color: rgb(255, 255, 255)
gtk-theme=headerbar-bg-color: rgb(48, 48, 48)
gtk-theme=headerbar-backdrop-color: rgb(36, 36, 36)
```

Basically, each `gtk-theme` line specifies a CSS variable name (minus the leading `--`) and the CSS color specification.

CSS variables could be added for many Ghostty config settings:

```
:root {
  --ghostty-foreground: rgb(255, 255, 255)
  --ghostty-background: rgb(40, 44, 53)
  --ghostty-palette-0: rgb(29, 31, 33)
  <etc>
}
```

That way they could be used in `gtk-theme` entries like this:

```
gtk-theme=headerbar-fg-color: var(--ghostty-foreground)
gtk-theme=headerbar-bg-color: var(--ghostty-background)
gtk-theme=headerbar-backdrop-color: oklab(from var(--ghostty-bg-color) calc(l * 0.9) a b / alpha)
```

This would result in CSS that looked like this.

```
:root {
  --ghostty-foreground: rgb(255, 255, 255)
  --ghostty-background: rgb(40, 44, 53)
  --ghostty-palette-0: rgb(29, 31, 33)
  --headerbar-fg-color: var(--ghostty-foreground)
  --headerbar-bg-color: var(--ghostty-background)
  --headerbar-backdrop-color: oklab(from var(--ghostty-bg-color) calc(l * 0.9) a b / alpha)
}
```

**Comment by jcollie** [comment]
NixOS runs terminal emulators through a couple basic tests using VMs:

https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/terminal-emulators.nix

**Comment by jcollie** [comment, marked as answer]
Fixed by #3806
